http/handler.go

```diff
@@ -139,6 +141,18 @@ func requestAuth(r *http.Request, req *logical.Request) *logical.Request {
 	return req
 }
 
+// Determines the type of the error being returned and sets the HTTP
```

Comment:
What happens if you have multiple types of errors generated? Which one trumps the other?